File: src/main/java/aml_org/examples/Parsing.java
package aml_org.examples;

import amf.client.AMF;
import amf.client.model.document.DialectInstance;
import amf.client.parse.Aml10Parser;

import java.util.concurrent.ExecutionException;

import static java.lang.System.out;

public class Parsing {

    //#parsing-example
    public static void main(String[] args) throws ExecutionException, InterruptedException {

        AMF.init().get();

        AMF.registerDialect("file://aml/music/dialect/playlist.yaml").get();

        DialectInstance model = (DialectInstance) new Aml10Parser().parseFileAsync("file://aml/music/playlist1.yaml").get();

        displayEncodedElement(model);
    }

    static void displayEncodedElement(DialectInstance model) {
        out.println("Encoded element:");
        out.println("  " + model.encodes().id());
        out.println("Class terms for element:");
        for (String type : model.encodes().getTypeUris()) {
            out.println(" - " + type);
        }
    }

}

File: src/main/js/Parsing.js
var amf = require("amf-client-js");
var AMF = amf.AMF;
var Aml10Parser = amf.Aml10Parser;

//#parsing-example
AMF.init()
    .then(function() {
        return AMF.registerDialect("file://aml/music/dialect/playlist.yaml")
    })
    .then(function() {
        new Aml10Parser()
            .parseFileAsync("file://aml/music/playlist1.yaml")
            .then(displayEncodedElement)
    });

function displayEncodedElement(model) {
    console.log("Encoded element:");
    console.log("  " + model.encodes.id);
    console.log("Class terms for element:");
    model.encodes.getTypeUris().forEach(function(type) {
        console.log(" - " + type);
    });
}